### ISSUE TITLE
Handling Unknown Concatenated Generic Paramspecs

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#4991]
**Describe the bug**
```python
from typing import Any, Callable, Generic, TypeVar

from typing_extensions import Concatenate, ParamSpec

P = ParamSpec("P")
InputT = TypeVar("InputT")


class CallFn(Generic[InputT, P]):
    def __init__(self, func: Callable[Concatenate[InputT, P], None], *args: P.args, **kwargs: P.kwargs) -> None:
        ...


foo: Any = ...
foo_call = CallFn(foo, y=2)
```

**To Reproduce**
This produces an error message of ,

`Param spec "P@CallFn" has no bound value`

when no error message is expected. 

**VS Code extension or command-line**
Checked both pylance and pyright 1.1.304.

**Additional context**
A more realistic example and one I actually encountered is foo is a function imported from an untyped library. I would expect some of strict unknown type rules to trigger, but I wouldn't expect basic type error to happen here. If foo's signature looks like def foo(x: int, y: int) then the line is safe.

The concatenate is also important. Simplifying example to remove Concatenate the error goes away.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.305, which I just published. It will also be included in a future release of pylance.
